skada/metrics.py

```diff
@@ -552,9 +552,25 @@ def _score(self, estimator, X, y, sample_domain=None):
         X, y, sample_domain = check_X_y_domain(X, y, sample_domain)
         source_idx = extract_source_indices(sample_domain)
 
-        backward_estimator = deepcopy(estimator)
+        # if estimator is NeuralNet
+        if not isinstance(estimator, BaseEstimator):
+            # The estimator is a deep model
+            from skorch import NeuralNet
 
-        y_pred_target = estimator.predict(X[~source_idx])
+            if isinstance(estimator, NeuralNet):
+                backward_estimator = clone(estimator)
+            else:
+                raise ValueError(
+                    "The estimator passed should be a scikit-learn estimator or "
```

Comment:
Why restrict to either 2 of them? It seems like anything that supports `deepcopy` should be fine here, right?

If we do have our own wrapper for `skorch.NeuralNet`, we just need `__deepcopy__` method in there. If no, it would be better to check if the estimator is `skorch.NeuralNet` to make `clone` or `deepcopy` otherwise. If the user passes something that is not `NeuralNet` **and** it doesn't support `deepcopy`, they will get `copy.Error` (I guess).